File: leetcode/0205-isomorphic-strings/isIsomorphic.test.ts
import cases from "./cases.json";
import isIsomorphic from "./isIsomorphic";

test("isomorphic strings", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = isIsomorphic(testCase.input.s, testCase.input.t);
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toBe(testCase.output);
  });
});

File: leetcode/0205-isomorphic-strings/isIsomorphic_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type IsIsomorphicCase struct {
	Input struct {
		S string
		T string
	}

	Output bool
}

func TestIsIsomorphicCase(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]IsIsomorphicCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := isIsomorphic(testCase.Input.S, testCase.Input.T)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
